quantcdn/drupal
# Added canonical redirects and fixed unpublish issue.

See details in Drupal.org issues:

https://www.drupal.org/project/quantcdn/issues/3339392
https://www.drupal.org/project/quantcdn/issues/3414473
https://www.drupal.org/project/quantcdn/issues/3414476

1. Get canonical redirects working for nodes/terms/files (didn't test multilingual).
2. Fixed redirects when aliases change.
3. Fixed issue with nodes not being unpublished.

Tested 1 and 2 for nodes/terms/files by:

1. Create entity, check canonical redirect is correct in dashboard.
2. Change alias, check canonical and other redirects are correct in dashboard.
3. Delete entity, check redirects are unpublished in dashboard.
4. Tested with tracking on (didn't have to run queue) and off (had to run queue).

Tested 3 for nodes by:

1. Create published node.
2. Unpublish node.
3. Check in dashboard that node is unpublished.
4. Tested with tracking on (didn't have to run queue) and off (had to run queue).
